File: Frontend/src/app/Model/Invoice.ts
import { Customer } from "./Customer";

export interface Invoice {
  id: number | undefined | null;
  invoiceNo: number | undefined | null;
  customer: Customer;
  customerEmail: string | undefined | null;
  business: string | undefined | null;
  sendLater: boolean | undefined | null;
  billingAddress: string | undefined | null;
  terms: string | undefined | null;
  invoiceDate: Date | undefined | null;
  dueDate: Date | undefined | null;
  balanceDue: number | null | undefined;
  invoiceProductDtoList: InvoiceProduct[];
  message: string | undefined | null;
  statement: string | undefined | null;
  status: boolean | undefined | null;
}
export interface InvoiceProduct {
  id: number | undefined | null;
  dateRow: Date | undefined | null;
  productRow: number | undefined | null;
  productName: string | undefined | null;
  type: string | undefined | null;
  description: string | undefined | null;
  qty: number | undefined | null;
  rate: number | undefined | null;
  amount: number | undefined | null;
  status: boolean | undefined | null;
}

File: src/main/java/com/PrintLab/model/Invoice.java
package com.PrintLab.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class Invoice {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(unique = true)
    private Long invoiceNo;
    private Long customer;
    private String customerEmail;
    private String business;
    private Boolean sendLater;
    private String billingAddress;
    private Double balanceDue;
    private String terms;
    private Date invoiceDate;
    private Date dueDate;
    @OneToMany(mappedBy = "invoice", cascade = CascadeType.ALL)
    private List<InvoiceProduct> invoiceProduct;
    private String message;
    private String statement;
    private Boolean status;

}
